### PR TITLE
TRAC-2 Contact Trigger Zip Code CCID

### PR DESCRIPTION
**Pull request by jeffweyhrauch**
FFLIB TRAC 2

### CODE REVIEW

File: src/classes/ContactHelper.cls [hunk 1]
```diff
@@ -0,0 +1,44 @@
+public class ContactHelper extends fflib_SObjectDomain implements IContact {
```
Comment (christiancoleman) [inline src/classes/ContactHelper.cls]: Please rename to **Contacts**

File: src/classes/ContactHelper.cls [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    public static fflib_ISObjectUnitOfWork uow = fflibext_Application.UnitOfWork.newInstance();
+
+    public static IContact newInstance(List<Contact> records){
+        return (IContact) fflibext_Application.Domain.newInstance(records);
```
Comment (christiancoleman) [inline src/classes/ContactHelper.cls]: Convert file to tabs instead of spaces

File: src/classes/ContactSelector.cls [hunk 3]
```diff
@@ -0,0 +1,27 @@
+public class ContactSelector extends fflib_SObjectSelector implements IContactSelector {
+
+	public static IContactSelector newInstance(){
```
Comment (christiancoleman) [inline src/classes/ContactSelector.cls]: Spaces to tab here too

Comment (christiancoleman) [reply src/classes/ContactSelector.cls]: Nevermind this one should be good. But double check to be sure.

File: src/classes/ContactServiceImpl.cls [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+				Set<String> zipPlus4 = new Set<String>();
+				Set<String> zip4 = new Set<String>();
+				Set<String> zip5 = new Set<String>();
+
+				List<Id> accountIdList = new List<Id>();
+				Map<Id,Account> idToAccountMap = new Map<Id,Account>();
+				for(Contact con : contactsToUpdate) {
+					accountIdList.add(con.AccountId);
+				}
+				List<Account> accountList = [SELECT Name,Id,BillingPostalCode,BillingCountry,Congressional_District__c,County_Name__c,State_Code__c from ACCOUNT where Id IN: accountIdList];
```
Comment (christiancoleman) [inline src/classes/ContactServiceImpl.cls]: Let's meet on switching this to using a selector. Set some time on my schedule

File: src/classes/IContact.cls [hunk 5]
```diff
@@ -0,0 +1,5 @@
+public interface IContact extends fflib_ISObjectDomain {
+
+    List<Contact> updateCCDLookupIdContact();
```
Comment (christiancoleman) [inline src/classes/IContact.cls]: Spaces to tabs

File: src/triggers/ContactTrigger.trigger [hunk 6]
```diff
@@ -0,0 +1,5 @@
+trigger ContactTrigger on Contact (before insert, before update) {
+
+    fflib_SObjectDomain.triggerHandler(ContactHelper.class);
```
Comment (christiancoleman) [inline src/triggers/ContactTrigger.trigger]: Spaces to tabs, also be sure to rename to Contacts.class here too

File: src/classes/UpdateCCDLookupIdTest.cls [hunk 7]
```diff
@@ -114,12 +114,199 @@ private class UpdateCCDLookupIdTest {
...
-			LIMIT 1
+				SELECT Id, County_Name__c
+				FROM Account
+				LIMIT 1
 		];
 
 		System.assertEquals(congressionalDistrict.County_Name__c, acc.County_Name__c, 'Account\'s county name does not match');
 	}
 
+	@isTest static void ContactZipcodeMatch() {
```
Comment (christiancoleman) [inline src/classes/UpdateCCDLookupIdTest.cls]: From the best practices document we need to convert these test method names:

Test methods should follow this format: testMethodThatYouAreTesting_StateBeforeTestAndWhatIsBeingSetup_ExpectedResult

https://github.com/InternationalTradeAdministration/developer-best-practices

Comment (christiancoleman) [reply src/classes/UpdateCCDLookupIdTest.cls]: Hey Jrock I just realized these aren't apart of the approved best practices yet. You don't need to them

File: src/classes/ContactSelector.cls (hunk 3 above)
Comment (jeffweyhrauch) [reply src/classes/ContactSelector.cls]: It displays as tabs in WIC and in my settings it has use tabs checked

**Pull request merged**